packages/markdown/loader.js

```diff
@@ -438,32 +460,109 @@ module.exports = async function demoLoader() {
-                // Avoid duplicates
-                if (!modules[variant].some((elem) => elem.module === relativeModuleID)) {
-                  modules[variant].push(moduleData);
+              try {
+                // We are only iterating trough an array that looks
+                // like this: ['JS', 'TS'], so  it is safe to await
+                // eslint-disable-next-line no-await-in-loop
+                raw = await fs.readFile(relativeModuleFilePath, {
+                  encoding: 'utf8',
+                });
+
+                const importedProcessedModuleIDs = new Set();
+                const importedProcessedModulesIDsParents = new Map();
+                // Find the relative paths in the relative module
+                if (relativeModuleID.startsWith('.')) {
+                  extractImports(raw).forEach((importModuleID) => {
+                    // detect relative import
+                    const importModuleIdWithExtension = detectRelativeImports(
+                      importModuleID,
+                      relativeModuleFilePath,
+                      variant,
+                      importModuleID,
+                    );
+                    if (importModuleID.startsWith('.')) {
+                      importedProcessedModuleIDs.add(importModuleIdWithExtension);
+                      importedProcessedModulesIDsParents.set(
+                        importModuleIdWithExtension,
+                        relativeModuleFilePath,
+                      );
+                    }
+                  });
+
+                  if (!addedModulesRelativeToModulePath.has(relativeModuleFilePath)) {
+                    const moduleData = { module: relativeModuleID, raw };
+                    updateRelativeModules(moduleData, variant);
+                    addedModulesRelativeToModulePath.add(relativeModuleFilePath);
+                  }
                 }
-              } else {
-                modules[variant] = [moduleData];
+                // iterate recursively over the relative imports
+                while (importedProcessedModuleIDs.size > 0) {
+                  for (const entry of importedProcessedModuleIDs) {
+                    if (entry.startsWith('.')) {
+                      const entryModuleFilePath = path.join(
+                        path.dirname(importedProcessedModulesIDsParents.get(entry)),
+                        entry,
+                      );
+
+                      // We are only iterating trough an array that looks
+                      // like this: ['JS', 'TS'], so  it is safe to await
+                      // eslint-disable-next-line no-await-in-loop
+                      const rawEntry = await fs.readFile(entryModuleFilePath, { encoding: 'utf8' });
+
+                      extractImports(rawEntry).forEach((importModuleID) => {
+                        // detect relative import
+                        const importModuleIdWithExtension = detectRelativeImports(
+                          entry,
```

Comment:
```suggestion
                          relativeModuleID
```

This (and the previous suggestion) affect the `relativeModules` map from being formed as (for an example where `ComboBox.js` has a relative import called `top100Films.js`, which has a further relative import called `moreFilms.js`)

<table>
<thead>
<tr>
<th>Malformed</th>
<th>Correct</th>
</tr>
</thead>
<tbody>
<tr>
<td>
<pre>
relativeModules Map(2) {
'ComboBox.js' => Map(2) {
'./top100Films.js' => [ 'JS' ],
'./top100Films.ts' => [ 'TS' ]
},
'./moreFilms' => Map(2) 
 { 
  './moreFilms.js' => [ 'JS' ], 
'  ./moreFilms.ts' => [ 'TS' ] 
  }
}
</pre>
</td>
<td>
<pre>
relativeModules Map(3) {
'ComboBox.js' => Map(2) {
  './top100Films.js' => [ 'JS' ],
  './top100Films.ts' => [ 'TS' ]
},
'./top100Films.js' => Map(1) 
  { './moreFilms.js' => [ 'JS' ] },
 './top100Films.ts' => Map(1) 
  { './moreFilms.ts' => [ 'TS' ] }
}
</pre>
</td>
</tr>
</tbody>
</table>

Reply:
Nice catch! Updated, looks good now!